helper/config/config.go

```diff
@@ -0,0 +1,28 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-sockaddr/template"
+)
+
+// ParseSingleIPTemplate is used as a helper function to parse out a single IP
+// address from a config parameter.
+func ParseSingleIPTemplate(ipTmpl string) (string, error) {
```

Comment:
Copied from https://github.com/hashicorp/go-secure-stdlib/blob/7166b1cdf6a6eb1311247e34feae17eaf7b657bf/listenerutil/parse.go#L374

Reply:
Instead of moving the copy of the library, I think we could just use the library? The upstream repo scopes each package to it's own module, so we wouldn't pull in much new.

Reply:
Hm... I'm kind of very-soft-against because the package pulls in stuff like their tlsutils package, etc. But it looks like at least they're pinned to the same version of go-sockaddr, so maybe that's ok?

Reply:
I've slept on it and I'm sold... let's pull in `go-secure-stdlib/listenrutil` here